src/bigint.rs

```diff
@@ -34,12 +32,12 @@ impl BigUint {
         let mut carry = 0u64;
 
         for chunk in self.chunks.iter_mut() {
-            carry = (carry << 32) | *chunk as u64;
-            *chunk = (carry / divider as u64) as u32;
-            carry %= divider as u64;
+            carry = (carry << 32) | u64::from(*chunk);
+            *chunk = (carry / u64::from(divider)) as u32;
+            carry %= u64::from(divider);
         }
 
-        if self.chunks[0] == 0 && self.chunks.len() > 0 {
+        if self.chunks[0] == 0 && self.chunks.is_empty() {
```

Comment:
Replaced `.len() > 0` (is NOT empty) with `.is_empty()`.

I'd suggest replacing the condition with `if let Some(0) = self.chunks.get(0) {` or `if self.chunks.get(0) == Some(0) {`, to be more declarative and have non-panicing bounds check in a single call.

Reply:
Haven't noticed this comment. This was the bug that I introduced that slowed the encoding by more than a third. The decoding is also slow by a third in some cases IIRC, but it's not related to that, so I'm still looking into it.